source/frontend/src/features/mapSideBar/acquisition/tabs/expropriation/form8/Form8PaymentItemsSubForm.tsx

```diff
@@ -0,0 +1,205 @@
+import styled from 'styled-components';
+
+import { LinkButton, StyledRemoveLinkButton } from '@/components/common/buttons';
+import { FastCurrencyInput, Select } from '@/components/common/form';
+import { YesNoSelect } from '@/components/common/form/YesNoSelect';
+import GenericModal from '@/components/common/GenericModal';
+import { SectionField } from '@/components/common/Section/SectionField';
+import * as API from '@/constants/API';
+import useLookupCodeHelpers from '@/hooks/useLookupCodeHelpers';
+import { getCurrencyCleanValue, stringToBoolean } from '@/utils/formUtils';
+
+import { Form8FormModel, Form8PaymentItemModel } from './models/Form8FormModel';
+
+export interface IForm8PaymentItemsSubFormProps {
+  form8Id: number | null;
+  formikProps: FormikProps<Form8FormModel>;
+  gstConstantPercentage: number;
+}
+
+export const Form8PaymentItemsSubForm: React.FunctionComponent<IForm8PaymentItemsSubFormProps> = ({
+  form8Id,
+  formikProps,
+  gstConstantPercentage,
+}) => {
+  const { values, setFieldValue } = useFormikContext<Form8FormModel>();
+  const [showModal, setShowModal] = useState(false);
+  const [rowToDelete, setRowToDelete] = useState<number | undefined>(undefined);
+
+  const { getOptionsByType } = useLookupCodeHelpers();
+  const paymentItemTypesOptions = getOptionsByType(API.PAYMENT_ITEM_TYPES);
+
+  const onUpdateGstApplicable = (index: number, gstOption: string): void => {
+    const isGstRequired = stringToBoolean(gstOption);
+    setFieldValue(`paymentItems[${index}].isGstRequired`, gstOption);
+    setAmountFields(index, isGstRequired, values.paymentItems[index].pretaxAmount);
+  };
+
+  const onPretaxAmountUpdated = (index: number, newValue: string): void => {
+    const isGstRequired = stringToBoolean(values.paymentItems[index].isGstRequired);
+    const cleanValue = getCurrencyCleanValue(newValue);
+
+    setAmountFields(index, isGstRequired, cleanValue);
+  };
+
+  const onTaxAmountUpdated = (index: number, newValue: string): void => {
+    let totalAmount = 0;
+    const preTaxAmount = values.paymentItems[index].pretaxAmount;
+    const cleanValue = getCurrencyCleanValue(newValue);
+
+    totalAmount = preTaxAmount + cleanValue;
+    setFieldValue(`paymentItems[${index}].totalAmount`, totalAmount);
+  };
+
+  const setAmountFields = (index: number, gstRequired: boolean, pretaxAmount: number): void => {
+    let totalAmount = 0;
+    let taxAmount = 0;
+
+    if (gstRequired) {
+      taxAmount = pretaxAmount * gstConstantPercentage;
+      totalAmount = taxAmount + pretaxAmount;
+    } else {
+      totalAmount = pretaxAmount;
+    }
+
+    setFieldValue(`paymentItems[${index}].taxAmount`, taxAmount);
+    setFieldValue(`paymentItems[${index}].totalAmount`, totalAmount);
+  };
+
+  return (
+    <FieldArray
+      name="paymentItems"
+      render={arrayHelpers => {
+        return (
+          <>
+            {values.paymentItems.map((item, index) => (
+              <div data-testid={`paymentItems[${index}]`}>
```

Comment:
you don't have  unique key on this child so react is probably throwing an error.

Reply:
updated